### PR TITLE
Experimental typescript support

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Broken out from #210, since that's likely to take a while.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "target": "es2020",
+    "typeRoots": [
+      "./node_modules/@types",
+      "./test/types"
+    ]
+  },
+  "include": [
+    "src/**/*"
+  ]
+}
```
Comment (jelly) [inline tsconfig.json:22]: One small nit:

We do this via an ugly? Dynamic include so it's logical it doesn't know but can we teach it about it?

image

Comment (jelly) [reply tsconfig.json:22]: Seems we can do this:

```json
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "jsx": "react",
    "lib": [
      "dom",
      "es2020"
    ],
    "paths": {
	    "*": ["*", "pkg/lib/*"]
    },
```

**Comment by allisonkarlitskaya** [comment]
@jelly this looks much better now, thanks for the tip

**Comment by jelly** [comment]
> @jelly this looks much better now, thanks for the tip

\o/

Also let's land this now, we can always improve on it in future PR's

**Review by jelly — APPROVED** [review]

**Comment by allisonkarlitskaya** [comment]
Should be better now.

**Review by jelly — APPROVED** [review]
Let's type it

**Pull request merged**